op-interop-mon/monitor/maintainer.go

```diff
@@ -0,0 +1,160 @@
+package monitor
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-interop-mon/metrics"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/locks"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+// JobUpdater can take cases and update them
+type JobUpdater interface {
+	UpdateJob(c Job)
+}
+
+type receiptClient interface {
+	BlockReceipts(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash) ([]*types.Receipt, error)
+}
+
+type Maintainer struct {
+	clients     locks.RWMap[eth.ChainID, receiptClient]
+	finders     locks.RWMap[eth.ChainID, Finder]
+	updaters    locks.RWMap[eth.ChainID, Updater]
+	newInbox    chan *Job
+	updateInbox chan *Job
+	closed      chan struct{}
+
+	jobMap   map[JobID]*Job
+	jobMapMu sync.RWMutex
```

Comment:
What do you think about having each updater store its own jobs? Then finders can talk directly to updaters, they only need the maintainer as a telephone directory.